File: Helpers/src/main/java/com/dg/controls/BouncyListView.java
package com.dg.controls;

import android.content.Context;
import android.util.AttributeSet;
import android.util.DisplayMetrics;
import android.widget.ListView;

/**
 * Created by Daniel Cohen Gindi (danielgindi@gmail.com)
 */
public class BouncyListView extends ListView {

    private Context context;
    private int maxYOverscrollDistance;

    public BouncyListView(Context context)
    {
        super(context);
        this.context = context;
        initBounceListView();
    }

    public BouncyListView(Context context, AttributeSet attrs)
    {
        super(context, attrs);
        this.context = context;
        initBounceListView();
    }

    public BouncyListView(Context context, AttributeSet attrs, int defStyle)
    {
        super(context, attrs, defStyle);
        this.context = context;
        initBounceListView();
    }

    private void initBounceListView()
    {
        setMaxYOverscrollDistance(200);
    }

    public void setMaxYOverscrollDistance(int maxYOverscrollDistance)
    {
        final DisplayMetrics metrics = context.getResources().getDisplayMetrics();
        final float density = metrics.density;

        this.maxYOverscrollDistance = (int) (density * maxYOverscrollDistance);
    }

    @Override
    protected boolean overScrollBy(int deltaX, int deltaY, int scrollX, int scrollY, int scrollRangeX, int scrollRangeY, int maxOverScrollX, int maxOverScrollY, boolean isTouchEvent)
    {
        return super.overScrollBy(deltaX, deltaY, scrollX, scrollY, scrollRangeX, scrollRangeY, maxOverScrollX, this.maxYOverscrollDistance, isTouchEvent);
    }

}

File: Helpers/src/main/java/com/dg/controls/BouncyListView.kt
package com.dg.controls

import android.content.Context
import android.util.AttributeSet
import android.widget.ListView

class BouncyListView : ListView
{
    private var maxYOverscrollDistance: Int = 0

    constructor(context: Context) : super(context)
    {
        initBounceListView()
    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)
    {
        initBounceListView()
    }

    constructor(context: Context,
                attrs: AttributeSet,
                defStyle: Int) : super(context, attrs, defStyle)
    {
        initBounceListView()
    }

    private fun initBounceListView()
    {
        setMaxYOverscrollDistance(200)
    }

    fun setMaxYOverscrollDistance(maxYOverscrollDistance: Int)
    {
        val metrics = context!!.resources.displayMetrics
        val density = metrics.density

        this.maxYOverscrollDistance = (density * maxYOverscrollDistance).toInt()
    }

    override fun overScrollBy(deltaX: Int,
                              deltaY: Int,
                              scrollX: Int,
                              scrollY: Int,
                              scrollRangeX: Int,
                              scrollRangeY: Int,
                              maxOverScrollX: Int,
                              maxOverScrollY: Int,
                              isTouchEvent: Boolean): Boolean
    {
        return super.overScrollBy(deltaX, deltaY, scrollX, scrollY, scrollRangeX, scrollRangeY, maxOverScrollX, this.maxYOverscrollDistance, isTouchEvent)
    }

}
